packages/predicate/src/predicate.ts

```diff
@@ -85,19 +81,18 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
     configurableConstants?: { [name: string]: unknown }
   ) {
     let predicateBytes = arrayify(bytes);
-    let predicateTypes: ReadonlyArray<JsonAbiFragmentType> | undefined;
-    let predicateInterface: Interface | undefined;
+    let abiInterface: Interface | undefined;
 
     if (jsonAbi) {
-      predicateInterface = new Interface(jsonAbi as JsonAbi);
-      const mainFunction = predicateInterface.fragments.find(({ name }) => name === 'main');
+      abiInterface = new Interface(jsonAbi);
+      const mainFunction = abiInterface.functions.main;
       if (mainFunction !== undefined) {
-        predicateTypes = mainFunction.inputs;
+        // predicateTypes = mainFunction.inputs;
       } else {
         logger.throwArgumentError(
           'Cannot use ABI without "main" function',
-          'Function fragments',
-          predicateInterface.fragments
+          'Abi functions',
+          abiInterface.functions
         );
       }
```

Comment:
Maybe this re-written if-else block could use some refactoring, since the new code doesn't do anything if `mainFunction` is not `undefined`. Should also remove that commented out line of code.